6_survival_analysis/nbconverted/survival_hit_groups.py

```diff
@@ -0,0 +1,704 @@
+    copy_status.index = copy_status.index.str[:12]
+        
+    # get groups from mutation/CNV data
+    if hits_classification == 'one':
+        id_clinical_df['is_mutated'] = (mutation_status | copy_status).astype(int)
+        id_clinical_df['is_mutated_alt'] = (mutation_status & copy_status).astype(int)
+    elif hits_classification == 'two':
+        id_clinical_df['is_mutated'] = (mutation_status & copy_status).astype(int)
+        id_clinical_df['is_mutated_alt'] = (mutation_status | copy_status).astype(int)
+        
+    return id_clinical_df
+
+
+# In[19]:
+
+
+# test for a selected example
+identifier = 'IDH1_LGG'
+cancer_classification = 'Oncogene'
+hits_classification = 'one'
+
+id_clinical_df = get_groups_for_gene_and_tissue(
+    identifier, cancer_classification, hits_classification)
+
+print(id_clinical_df.shape)
+print(id_clinical_df.is_mutated.sum(), id_clinical_df.is_mutated_alt.sum())
+print(id_clinical_df.isna().sum()) 
+id_clinical_df.head()
+
+
+# In[20]:
+
+
+# plot groups
+sns.set({'figure.figsize': (12, 6)})
+sns.set_style('whitegrid')
+
+fig, axarr = plt.subplots(1, 2)
+
+def plot_id(identifier, id_clinical_df):
+    
+    gene, tissue = identifier.split('_')
+
+    for ix, mut_col in enumerate(['is_mutated', 'is_mutated_alt']):
+
+        ax = axarr[ix]
+
+        n_mutant = id_clinical_df[mut_col].sum()
+        n_wildtype = id_clinical_df.shape[0] - n_mutant
+
+        for is_mutated in sorted(id_clinical_df[mut_col].unique()):
+            mask_mutated = (id_clinical_df[mut_col] == is_mutated)
+            time_treatment, survival_prob_treatment = kaplan_meier_estimator(
+                id_clinical_df['status'][mask_mutated],
+                id_clinical_df['time_in_days'][mask_mutated]
+            )
+
+            def get_label(gene, is_mutated, n_mutant, n_wildtype):
+                if is_mutated:
+                    return '{} mutant (n={})'.format(gene, n_mutant)
+                else:
+                    return '{} wild-type (n={})'.format(gene, n_wildtype)
+
+            # TODO: confidence intervals
+            ax.step(time_treatment, survival_prob_treatment, where="post",
+                    label=get_label(gene, is_mutated, n_mutant, n_wildtype))
+
+        ax.legend()
+
+        def get_n_hits(hits_class, ix):
+            hits_desc = (['one', 'two'] if hits_class == 'one' else ['two', 'one'])
+            return hits_desc[ix]
+
+        ax.set_title('Survival for {}-hit classification'.format(
+            get_n_hits(hits_classification, ix)))
+
+        # hypothesis testing using log-rank test
+        y = Surv.from_dataframe('status', 'time_in_days', id_clinical_df)
+        chisq, p_val = compare_survival(y, id_clinical_df[mut_col].values)
+        print(mut_col, 'chisq = {:.4f}'.format(chisq), 'p = {:.4e}'.format(p_val))
+
+    plt.suptitle('Comparing mutation classes for {}'.format(identifier))
+    
+plot_id(identifier, id_clinical_df)
```

Comment:
Is it really possible to understand significance using a kaplan-meier curve with such a small sample size as in the two-hits in this example?

Reply:
Just saw that you discussed this further down, nevermind.

Reply:
Yeah, all of this is confounded by sample size, which is always going to be smaller for the two-hit samples. Adding the cutoffs for number of mutated samples and p-values helps a bit, but it's not a perfect solution.